### ISSUE TITLE
List of small updates for the library

### ISSUE THREAD

**Issue by thunderbiscuit** [bitcoindevkit/bdk-jni#38]
While playing around the library for #37, I made a list of small upkeep items that would be good to tackle in order to bring the library up to date. This is an issue to keep track of them and discuss some of the specifics.

### Tooling
- [x] 1. Gradle wrapper is a full version behind (`6.6.1`, latest is `7.1.1`)
- [x] 2. Gradle plugin is a full version behind (`3.6.1`, latest is `4.2.2`)
- [x] 3. Kotlin version is `1.4.0`, latest is `1.5.21`
- [x] 4. Kotlin Gradle plugin should be updated to match Kotlin version

### Files Layout
- [x] 5. The fact that bdk-jni relies on a Rust crate to build a native library file which the `bdk-jvm` and `bdk-android` libraries rely on is, IMO, very much an internal implementation detail. Yet the crate sits at the root of the repository, which I find is more confusing than anything, because it brings the `Cargo.toml` and `Cargo.lock` files and the `src` directory right at the top. I propose we pull the crate one layer down into its own directory.

### Documentation
- [x] 6. The documentation is not yet very explicit when it comes to building _just_ the `bdk-jvm` or the `bdk-android` libraries. I think we should expect future user to arrive to the repository with the mind to build _one_ of them (their need is probably either JVM or Android, not both at the same time). The fact is that `bdk-android` indeed requires `bdk-jvm` to be built and published for it to work, but that's not obvious from the outset and should should be explained (as a new user, I'd be looking at the docs thinking: "why are they getting me to compile everything for all architectures and all the Android stuff when I really just want to build the `bdk-jvm` library for a linux cloud server?"). The JVM library can be build by itself and does not require `bdk-android`. The docs are still not very clear on this.
- [x] 7. The docs mention that you need JDK 14 or later to build, but I've been using 11.0.11 with success. Java 11 is the latest LTS (and more popular). I think if we can indeed build with 11 (I wonder if I might have not tried everything or if there was a particular reason for JDK 14, let me know), we should mention it in the docs.
- [x] 8. The docs should mention the Rust version required to build.

### Other
- [x] 9. The license should be updated to be MIT/Apache 2.0

**Comment by thunderbiscuit** [comment]
Feel free to propose any other upkeep ideas. Also, I'm wondering if it makes sense to have a single PR for all of this (once the items have been discussed and agreed upon) or if they should maybe be broken down into 3ish PRs.

**Comment by notmandatory** [comment]
I think each of those headings (tool versions, layout, docs, license) could be done in their own PRs, but don't really need a separate PR for each individual checkbox item.

**Comment by thunderbiscuit** [comment]
The PR addressing the tooling updates is ready to go, but relies on #39. Once that's merged I'll open that PR on the `master` branch.

https://github.com/thunderbiscuit/bdk-jni/pull/1

**Comment by thunderbiscuit** [comment]
Task number 9 is done and sitting on branch [`new-licenses`](https://github.com/thunderbiscuit/bdk-jni/tree/new-licenses), ready for a PR once the `tooling/upkeep` PR is merged.

**Comment by notmandatory** [comment]
Completed with #40
